python/lsst/sims/catUtils/utils/LightCurveGenerator.py

```diff
@@ -0,0 +1,496 @@
+        else:
+            obs_list = []
+            for bp in bandpass:
+                sub_list = self._generator.getObservationMetaData(fieldRA=ra,
+                                                                  fieldDec=dec,
+                                                                  telescopeFilter=bp,
+                                                                  expMJD=expMJD,
+                                                                  boundLength=1.75)
+                obs_list += sub_list
+
+        if len(obs_list) == 0:
+            print("No observations found matching your criterion")
+            return None
+
+        # Group the OpSim pointings so that all of the pointings centered on the same
+        # point in the sky are in a list together (this will allow us to generate the
+        # light curves one pointing at a time without having to query the database for
+        # the same results more than once.
+        tol = 1.0e-12
+
+        obs_groups = []  # a list of list of the indices of the ObservationMetaDatas
+                         # in obs_list.  All of the ObservationMetaData in
+                         # obs_list[i] will point to the same point on the sky.
+
+        mjd_groups = []  # a list of lists of the MJDs of the ObservationMetaDatas
+                         # so that they can be sorted into chronological order before
+                         # light curves are calculated
+
+        for iobs, obs in enumerate(obs_list):
+            group_dex = -1
+
+            for ix, obs_g in enumerate(obs_groups):
+                dd = haversine(obs._pointingRA, obs._pointingDec,
+                               obs_list[obs_g[0]]._pointingRA, obs_list[obs_g[0]]._pointingDec)
+                if dd < tol:
+                    group_dex = ix
+                    break
+
+            if group_dex == -1:
+                obs_groups.append([iobs])
+                mjd_groups.append([obs_list[iobs].mjd.TAI])
+            else:
+                obs_groups[group_dex].append(iobs)
+                mjd_groups[group_dex].append(obs_list[iobs].mjd.TAI)
+
+        # rearrange each group of ObservationMetaDatas so that they
+        # appear in chronological order by MJD
+        obs_groups_out = []
+        for ix, (grp, mjd) in enumerate(zip(obs_groups, mjd_groups)):
+            oo = np.array(grp)
+            mm = np.array(mjd)
+            dexes = np.argsort(mm)
+            obs_groups_out.append([obs_list[ii] for ii in oo[dexes]])
+
+        return obs_groups_out
+
+    def _get_query_from_group(self, grp, chunk_size):
+        """
+        Take a group of ObervationMetaData that all point to the same region
+        of the sky.  Query the CatSim database for all of the celestial objects
+        in that region, and return it as an iterator over database rows.
+        """
+
+        cat = self._lightCurveCatalogClass(self._catalogdb, obs_metadata=grp[0])
+
```

Comment:
I believe so.  That is where InstanceCatalog verifies that your CatalogDBObject class contains all of the raw columns needed to generate your getter-calculated columns.